zerver/data_import/slack.py

```diff
@@ -314,7 +324,11 @@ def get_user_email(user: ZerverFieldsT, domain_name: str) -> str:
             slack_bot_name = user['profile']['first_name']
         else:
             raise AssertionError("Could not identify bot type")
-        return slack_bot_name.replace("Bot", "").replace(" ", "") + f"-bot@{domain_name}"
+        bot_email = slack_bot_name.replace("Bot", "").replace(" ", "") + f"-bot@{domain_name}"
+        if bot_email_list is not None:
+            if bot_email in bot_email_list:
+                return slack_bot_name.replace("Bot", "").replace(" ", "") + str(bot_id_count) + f"-bot@{domain_name}"
```

Comment:
Why do we need `bot_email_list`?  I would have imagined a simpler implementation could be entirely local to this function and just maintain a hash table mapping Slack bot IDs to their emails.

Reply:
@timabbott I created <code>bot_email_list</code> to keep track of <code>bot_emails</code> between different function calls .
By your hash table approach do you mean to create a dictionary of <code>bot_id</code> and <code>bot_email</code> 
beforehand and then pass it to this function .Check if emails clash and make change if necessary. Please tell if I have understood this correctly then I would make the change.